redbot/core/i18n.py

```diff
@@ -160,6 +171,32 @@ def _add_translation(self, untranslated, translated):
             self.translations[untranslated] = translated
 
 
+@functools.lru_cache()
+def _get_babel_locale(red_locale: str) -> babel.core.Locale:
+    supported_locales = babel.localedata.locale_identifiers()
+    try:  # Handles cases where red_locale is already Babel supported
+        babel_locale = Locale(*babel.parse_locale(red_locale))
+    except (ValueError, babel.core.UnknownLocaleError):
+        try:
+            babel_locale = Locale(*babel.parse_locale(red_locale, sep="-"))
+        except (ValueError, babel.core.UnknownLocaleError):
+            # ValueError is Raised by `parse_locale` when an invalid Locale is given to it
+            # Lets handle it silently and default to "en_US"
+            try:
+                # Try to find a babel locale that's close to the one used by red
+                babel_locale = Locale(Locale.negotiate([red_locale], supported_locales, sep="-"))
+            except (ValueError, TypeError, babel.core.UnknownLocaleError):
+                # If we fail to get a close match we will then default to "en_US"
+                babel_locale = Locale("en", "US")
+    return babel_locale
+
+
+def get_babel_locale(locale: Optional[str] = None) -> babel.core.Locale:
```

Comment:
Could you add a docstring for this function? :)

Reply:
![image](https://user-images.githubusercontent.com/27962761/63760362-134e9c00-c8b7-11e9-8258-58673caf481b.png)
Hows this?